packages/fast-usdc/src/cli/lp-commands.js

```diff
@@ -0,0 +1,161 @@
+    .description('Deposit USDC into pool')
+    .addHelpText(
+      'after',
+      '\nPipe the STDOUT to a file such as deposit.json, then use the Agoric CLI to broadcast it:\n  agoric wallet send --offer deposit.json --from gov1 --keyring-backend="test"',
+    )
+    .requiredOption('--amount <number>', 'USDC amount', parseDecimal)
+    .option('--offerId <string>', 'Offer id', String, `lpDeposit-${now()}`)
+    .action(async opts => {
+      vskP ||= loadVsk();
+      const vsk = await vskP;
+      /** @type {Brand<'nat'>} */
+      // @ts-expect-error it doesnt recognize usdc as a Brand type
+      const usdc = vsk.agoricNames.brand.USDC;
+      assert(usdc, 'USDC brand not in agoricNames');
+
+      const usdcAmount = parseUSDCAmount(opts.amount, usdc);
+
+      /** @type {OfferSpec} */
+      const offer = {
+        id: opts.offerId,
+        invitationSpec: {
+          source: 'agoricContract',
+          instancePath: ['fastUsdc'],
+          callPipe: [['makeDepositInvitation', []]],
+        },
+        proposal: {
+          give: {
+            USDC: usdcAmount,
+          },
+        },
+      };
+
+      /** @type {ExecuteOfferAction} */
+      const bridgeAction = {
+        method: 'executeOffer',
+        offer,
+      };
+
+      outputActionAndHint(bridgeAction, { stderr, stdout }, vsk.marshaller);
+    });
+
+  program
+    .command('withdraw')
+    .description("Withdraw USDC from the LP's pool share")
+    .addHelpText(
+      'after',
+      '\nPipe the STDOUT to a file such as withdraw.json, then use the Agoric CLI to broadcast it:\n  agoric wallet send --offer withdraw.json --from gov1 --keyring-backend="test"',
+    )
+    .requiredOption('--amount <number>', 'USDC amount', parseDecimal)
+    .option('--offerId <string>', 'Offer id', String, `lpWithdraw-${now()}`)
+    .action(async opts => {
+      vskP ||= loadVsk();
+      const vsk = await vskP;
+
+      /** @type {Brand<'nat'>} */
+      // @ts-expect-error it doesnt recognize FastLP as a Brand type
+      const poolShare = vsk.agoricNames.brand.FastLP;
+      assert(poolShare, 'FastLP brand not in agoricNames');
+
+      /** @type {Brand<'nat'>} */
+      // @ts-expect-error it doesnt recognize usdc as a Brand type
+      const usdc = vsk.agoricNames.brand.USDC;
+      assert(usdc, 'USDC brand not in agoricNames');
+
+      const usdcAmount = parseUSDCAmount(opts.amount, usdc);
+
+      /** @type {import('../types.js').PoolMetrics} */
+      // @ts-expect-error it treats this as "unknown"
+      const metrics = await vsk.readPublished('fastUsdc');
+      const fastLPAmount = ceilDivideBy(usdcAmount, metrics.shareWorth);
+
+      /** @type {OfferSpec} */
+      const offer = {
+        id: opts.offerId,
+        invitationSpec: {
+          source: 'agoricContract',
+          instancePath: ['fastUsdc'],
+          callPipe: [['makeWithdrawInvitation', []]],
+        },
+        proposal: {
+          give: {
+            PoolShare: fastLPAmount,
+          },
```

Comment:
let's include the want too.

I suggest using `USDCProposalShapes['withdraw']` for a static check.
You could even do an early dynamic check (see `makeProposalShapes`).

Reply:
Done